### ISSUE TITLE
Better control of Electron binary branding - less changes, smaller partial updates

### ISSUE THREAD

**Issue by SinusPi** [electron-userland/electron-builder#4490]
* app-builder-lib v21.2.0: 

* any under Windows

At every build, Electron.exe is branded by `app-builder rcedit` with various strings and/or icons. One possibly unwanted side-effect of this feature is that the 65MB .exe file compresses slightly differently, which results in a scattershot of changes in the resulting final package's blockmap.

It would be great if the branding could be controlled - for example branding the .exe with just the major version number, thus letting it be unchanged between minor or patch updates.

My current solution is to use metadata.shortVersion (if present) instead of appInfo.buildVersion or appInfo.getVersionInWeirdWindowsForm(), but I'm sure it can be polished. If the idea itself is acceptable, I'll gladly produce a pull request.

**Comment by SinusPi** [comment]
I ended up forking the project and making the necessary changes. Now I can set "shortVersion":"1.x" and "shortVersionWindows":"1.0.0.0" in my package.json file, to have branding lessened, both in the Electron.exe branding, and in the NSIS Uninstaller, if it's in use. Happy to provide a pull request, though so far I made it based on 21.2.0, the version I had in my modules.

As a side note - to help with reducing update sizes, I also set build.nsis.uninstallDisplayName to a static string (otherwise my uninstaller changes with every build). Also, I modified my ASAR configuration to pull my own code bundles out of the ASAR file, and even before that happens I pad my .js* files with spaces to a set size, so that the ASAR isn't changed at all between small code tweaks, not even file length entries change anymore. Outcome: replacing my whole code base is an 800 kB update, down from around 8 MB.
